### PR TITLE
feat: add interior handle ledges and fillet label tab support (#1105)

### LINKED ISSUE #1105: Wall cutout shape - handles?
**Problem**
What problem does this solve?
Often I like to take my bins out of a drawer to use them and put them back. It's difficult to pull the bins out when their sides are same-height and pushed up against each other.

**Proposed Solution**
Describe your idea.
I think holes on the sides of the bin to be used as handles (rather than just scoops/u-shapes) would fix this problem very well, and it would be great to have this as an alternate wall cutout shape.

### PR DESCRIPTION
**Pull request by andymai**
## Summary

- Adds interior handle ledges — small shelves with concave fillet supports on bin inner walls for gripping bins out of drawers
- Adds "Fillet" as a third label tab support option, sharing the concave arc profile with handles
- New "Handles" section in the Walls tab with per-side toggles (F/B/L/R), width/depth/fillet radius controls
- Constraint system blocks handles on slotted and solid bins
- Back wall handle auto-suppressed when label tabs are active
- Full i18n support across all 7 locales

Closes #1105

## Implementation

- `HandleConfig` type with per-side toggles on `BinParams`
- `handleBuilder.ts` — geometry builder (shelf plate + fillet support per wall)
- `filletProfile.ts` — shared concave quarter-circle profile (used by both handles and label tabs)
- Pipeline integration in `featuresStage.ts` as additive `fuseTargets`
- `HandleSection` UI component following `WallCutoutsSection` patterns
- Migration backfills defaults for old saved designs

## Test plan

- [ ] 8 handle builder unit tests (null cases, back suppression, edge cases: min depth/width, narrow bin clamping, fillet overflow)
- [ ] 4 label tab builder tests (bracket, solid, fillet, disabled)
- [ ] 3 scenario tests (handles + sides, handles + label tabs, handles + wall cutouts)
- [ ] 2 migration tests (backfill defaults, nested side merging)
- [ ] 4 HandleSection component tests
- [ ] 3 useHandleSection hook tests
- [ ] Visual verification: enable handles in bin designer, verify ledges appear on correct walls in 3D preview
- [ ] Verify fillet concavity direction (arc should curve inward toward wall-shelf junction)

### CODE REVIEW

File: src/features/generation/worker/generators/handleBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      .lineTo([handleWidth, -effectiveDepth])
+      .lineTo([0, -effectiveDepth])
+      .close();
+    const shelf = sketch(shelfDrawing, 'XY', 0).extrude(shelfThickness);
+
+    // -- Fillet support: concave quarter-circle under the shelf --
+    const filletHeight = Math.min(effectiveFilletR, interiorHeight - shelfThickness);
+
+    let handleSolid: Shape3D;
+
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/generation/worker/generators/handleBuilder.ts:102]: Fixed in commit 08991eb — removed the erroneous `-filletHeight` Z translation.

File: src/features/bin-designer/components/panel/HandleSection/HandleSection.test.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      params: { ...DEFAULT_BIN_PARAMS },
+      ui: { ...DEFAULT_UI_STATE },
+    });
+  });
+
+  it('renders handles toggle', () => {
+    render(<HandleSection />);
+    const labels = screen.getAllByText('Handles');
+    expect(labels.length).toBeGreaterThanOrEqual(1);
+  });
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/bin-designer/components/panel/HandleSection/HandleSection.test.tsx:19]: Panel section tests follow the `WallCutoutsSection.test.tsx` pattern which uses `useDesignerStore.setState()`. The `resetAllStores()` pattern is for full page-level tests (e.g. `DesignerPage.test.tsx`). No change needed.

File: src/features/bin-designer/components/panel/HandleSection/useHandleSection.test.ts [hunk 3]
```diff
@@ -0,0 +1,186 @@
+import { describe, it, expect, beforeEach } from 'vitest';
+import { renderHook, act } from '@testing-library/react';
+import { useHandleSection } from './useHandleSection';
+import { useDesignerStore } from '@/features/bin-designer/store';
+import { DEFAULT_BIN_PARAMS } from '@/features/bin-designer/constants';
+
+describe('useHandleSection', () => {
+  beforeEach(() => {
+    useDesignerStore.setState({
+      params: { ...DEFAULT_BIN_PARAMS },
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/bin-designer/components/panel/HandleSection/useHandleSection.test.ts:10]: Same as above — panel section tests use `setState()`, matching the established pattern in `WallCutoutsSection.test.tsx`.

File: src/features/generation/worker/generators/handleBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  if (!params.handles.enabled) return null;
+
+  const { depth, width, filletRadius } = params.handles;
+  const shelfThickness = Math.max(wallThickness, MIN_SHELF_THICKNESS);
+
+  const walls: readonly WallDef[] = [
+    { side: 'front', wallSpan: innerW, depthSpan: innerD, x: 0, y: -innerD / 2, rotateZ: 180 },
+    { side: 'back', wallSpan: innerW, depthSpan: innerD, x: 0, y: innerD / 2, rotateZ: 0 },
+    { side: 'left', wallSpan: innerD, depthSpan: innerW, x: -innerW / 2, y: 0, rotateZ: 270 },
+    { side: 'right', wallSpan: innerD, depthSpan: innerW, x: innerW / 2, y: 0, rotateZ: 90 },
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/generation/worker/generators/handleBuilder.ts]: Added clarifying comment about wall positioning and rotation axis mapping.

File: src/features/bin-designer/utils/validation.ts [hunk 5]
```diff
@@ -181,11 +181,14 @@ export function validateBinParams(params: BinParams): Result<BinParams, Designer
 
   // Label tab validation
   if (params.label.enabled) {
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- runtime guard for external data
-    if (params.label.support !== 'bracket' && params.label.support !== 'solid') {
+    if (
+      params.label.support !== 'bracket' &&
+      params.label.support !== 'solid' &&
+      params.label.support !== 'fillet'
+    ) {
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/bin-designer/utils/validation.ts:189]: Fixed — restored eslint-disable as block scope to cover the multi-line condition.

File: src/features/generation/worker/generators/pipeline/stages/featuresStage.ts [hunk 6]
```diff
@@ -209,6 +210,32 @@ export const featuresStage: PipelineStage = {
       );
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/generation/worker/generators/pipeline/stages/featuresStage.ts:210]: The pipeline's `dim.solid` early return at line 91 already prevents handles from generating in solid mode. The solid constraint at the UI layer is the primary guard; the pipeline's early return is the safety net. No additional guard needed.

File: src/features/bin-designer/components/panel/HandleSection/HandleSection.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+      </div>
+
+      {/* Width stepper */}
+      <div className="flex items-end gap-2">
+        <div className="flex-1 min-w-0">
+          <span className="mb-1 block text-xs text-content-tertiary">
+            {t('binDesigner.handles.width')}
+          </span>
+          <StepperControl
+            value={handles.width}
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/bin-designer/components/panel/HandleSection/HandleSection.tsx]: Fixed — added `(%)` and `(mm)` unit suffixes to the stepper labels.

File: src/features/bin-designer/components/panel/HandleSection/HandleSection.test.tsx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+  it('shows disabled reason for slotted bins', () => {
+    useDesignerStore.setState({
+      params: { ...DEFAULT_BIN_PARAMS, style: 'slotted' },
+    });
+
+    render(<HandleSection />);
+    expect(screen.getByText(/Not available/)).toBeDefined();
+  });
+});
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/bin-designer/components/panel/HandleSection/HandleSection.test.tsx:58]: Fixed — added solid bin constraint test.

File: src/features/bin-designer/components/panel/HandleSection/useHandleSection.ts [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+      updateHandles({ filletRadius });
+    },
+    [updateHandles]
+  );
+
+  const handleWidthMm = useMemo(() => {
+    const outerW = width * GRIDFINITY.GRID_SIZE - GRIDFINITY.TOLERANCE;
+    const innerW = outerW - 2 * wallThickness;
+    return Math.round(innerW * (handles.width / 100) * 10) / 10;
+  }, [width, wallThickness, handles.width]);
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/bin-designer/components/panel/HandleSection/useHandleSection.ts]: Fixed — `handleWidthMm` now accounts for both `innerW` (front/back) and `innerD` (left/right). When both orientations are enabled, it shows the minimum (most constrained) span.

File: src/features/generation/worker/generators/filletProfile.ts [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  const sagitta = safeR * (1 - 1 / Math.sqrt(2));
+
+  // Draw: start at shelf tip, go right along shelf to origin, down the wall
+  // to bottom, then arc back to shelf tip (concave, curving toward origin).
+  //
+  // sagittaArc takes relative dx, dy from current point to endpoint.
+  // From (0, -safeR) to (-depth, 0): dx = -depth, dy = safeR.
+  // Positive sagitta curves left of the travel direction. From bottom-right
+  // to top-left, "left" is toward the origin — which is the concave direction
+  // we want.
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/generation/worker/generators/filletProfile.ts]: Fixed — corrected the geometric reasoning in the sagitta direction comment.

File: src/features/bin-designer/components/panel/WallsSection/WallsSection.tsx [hunk 11]
```diff
@@ -39,6 +40,9 @@ export function WallsSection() {
       <div className="pt-3 border-t border-stroke-subtle/50">
         <WallCutoutsSection />
       </div>
+      <div className="pt-3 border-t border-stroke-subtle/50">
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/bin-designer/components/panel/WallsSection/WallsSection.tsx:43]: README updates are advisory per the pre-commit reminder. The bin-designer README doesn't document individual features at this level of detail — it covers architecture and gotchas. The handle feature follows established patterns and doesn't introduce new architectural concepts.

File: src/features/bin-designer/components/panel/HandleSection/useHandleSection.ts [hunk 12]
```diff
@@ -0,0 +1,111 @@
...
+    },
+    [updateHandles]
+  );
+
+  const setDepth = useCallback(
+    (depth: number) => {
+      updateHandles({ depth });
+    },
+    [updateHandles]
+  );
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/features/bin-designer/components/panel/HandleSection/useHandleSection.ts:64]: This matches the established codebase pattern — `useWallCutoutsSection.ts` uses the same `(side: WallSide, width: number)` naming convention. The callback parameter scoping is clear and consistent.
